misc pom/test/resource improvements

ryan-williams opened the pull request:
these happened while i was doing #1139; individual commits should be reasonably self-explanatory

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1440/<h2>Build result: FAILURE</span></h2>GitHub pull request #1142 of commit 3e02e9bfb4c40359c8e1d61cd3608e07b6dd407f automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > /home/jenkins/git2/bin/git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git -c core.askpass=true fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1142/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a --contains 3722878646d30b56dcaa16c9624bc0467db0486c # timeout=10 > /home/jenkins/git2/bin/git rev-parse remotes/origin/pr/1142/merge^{commit} # timeout=10Checking out Revision 3722878646d30b56dcaa16c9624bc0467db0486c (origin/pr/1142/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f 3722878646d30b56dcaa16c9624bc0467db0486cFirst time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

ryan-williams commented:
Some problem with where test-coverage reports are getting written after I changed directory configs from `surefire-reports` to `scalatest-reports`; I assumed that was find to just do everywhere but maybe something not in this repo has `surefire-reports` hard-coded? Looking…

ryan-williams commented:
The end of the log seems to get through the end of `jenkins-test` script successfully, so I can't tell what it's trying to do when the failure occurs:

```
+ echo 'All the tests passed'
All the tests passed
echo
+ echo

Recording test results
ERROR: Step ?Publish JUnit test result report? failed: No test report files were found. Configuration error?
Publishing Scoverage XML and HTML report...
Setting commit status on GitHub for https://github.com/bigdatagenomics/adam/commit/3722878646d30b56dcaa16c9624bc0467db0486c
Finished: FAILURE
```

fnothaft commented:
@ryan-williams IIRC, Jenkins has the test report paths hardcoded. Let me take a look at the config this AM!

heuermh commented:
Looks reasonable to me.

With `scala.version.prefix` will need to confirm the `move_to...` scripts still work correctly.

fnothaft commented:
Jenkins, test this please.

fnothaft commented on src/test/resources/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# We blanket-ignore .bam files in the top-level of the repo, but want to undo that here.
+!*.bam
```
Nifty! Didn't know you could do that.

ryan-williams replied:
Yea this caused me some problems when I tried to `git mv adam-core/* ./` to put all of adam-core in the top directory (after `git rm`ing the other modules); `sorted.bam*` did not come with that move since they were `.gitignore`d (despite having been `git add`ed to the repo at some point, maybe before they became `.gitignore`d?), so I was just missing them in my builds, which took me a while to figure out.

In any case, yea `.gitignore` "whitelists" can be useful in addition to the way it's usually used.

fnothaft replied:
Yeah originally we had a prohibition on adding binary files, but we've relaxed that over time. Thanks for patching this!

fnothaft commented:
Also, I have done the `surefire_reports` --> `scalatest_reports` path change in Jenkins, so that should be good to go.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1441/<h2>Build result: FAILURE</span></h2>GitHub pull request #1142 of commit 3e02e9bfb4c40359c8e1d61cd3608e07b6dd407f automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > /home/jenkins/git2/bin/git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git -c core.askpass=true fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1142/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a --contains 3722878646d30b56dcaa16c9624bc0467db0486c # timeout=10 > /home/jenkins/git2/bin/git rev-parse remotes/origin/pr/1142/merge^{commit} # timeout=10Checking out Revision 3722878646d30b56dcaa16c9624bc0467db0486c (origin/pr/1142/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f 3722878646d30b56dcaa16c9624bc0467db0486cFirst time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

ryan-williams commented:
OK, just pushed an attempt at updating the `move_to` scripts to use `2.11.8` and `2.10.6`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1442/<h2>Build result: FAILURE</span></h2>GitHub pull request #1142 of commit 372b573b13976e5eb18f95cc185d38cd7b98d116 automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > /home/jenkins/git2/bin/git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git -c core.askpass=true fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1142/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a --contains ee024234052b2bd21ebcbadd15ff96785320fdcc # timeout=10 > /home/jenkins/git2/bin/git rev-parse remotes/origin/pr/1142/merge^{commit} # timeout=10Checking out Revision ee024234052b2bd21ebcbadd15ff96785320fdcc (origin/pr/1142/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f ee024234052b2bd21ebcbadd15ff96785320fdccFirst time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

fnothaft commented on pom.xml:
```diff
@@ -268,7 +268,8 @@
         <configuration>
-          <scalaVersion>2.10.4</scalaVersion>
+          <scalaVersion>${scala.version}</scalaVersion>
+          <recompileMode>incremental</recompileMode>
```
We'd removed incremental in the past. I'd need to dig through the commit history to find the exact reason, but it has something to do with causing transient build issues. The Jenkins build just failed with an issue related to such. Alternatively, we could do `-DrecompileMode=all` in `scripts/jenkins-test` as suggested by https://github.com/davidB/scala-maven-plugin/issues/185. I'm open to either. It'd be nice to have faster Scala compile times in the general case.

ryan-williams replied:
Interesting; so incremental compilation in Scala 2.11 either doesn't work correctly at all, or only works in SBT?

Maybe a better option would be to enable a different Maven profile in the presence of a jenkins environment variable?

That way we would get incremental compiles locally, which would be nice, unless it's so broken we don't think we want that.

ryan-williams commented:
Just following up here, the last build failed on 2.6.0 x 1.5.2 x 2.10 with:

```
[error] Required file not found: sbt-interface.jar
[error] See zinc -help for information about locating necessary files
```

and 2.6.0 x 1.5.2 x 2.11 with:

```
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.bdgenomics.adam:adam-core_2.11:0.19.1-SNAPSHOT (/home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.6.0/SCALAVER/2.11/SPARK_VERSION/1.5.2/label/centos/adam-core/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.parquet:parquet-scala_2.10:jar is missing. @ line 155, column 17
[ERROR] 
```

I don't have obvious leads about either one.

heuermh commented:
The `parquet-scala_2.10` dependency is strange as there is no 2.11 version.
http://search.maven.org/#search%7Cga%7C1%7Cparquet-scala

The `move_to_scala_2.11.sh` script has a workaround, maybe something went wrong
https://github.com/bigdatagenomics/adam/blob/master/scripts/move_to_scala_2.11.sh#L6

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1517/<h2>Build result: FAILURE</span></h2>GitHub pull request #1142 of commit ef8e35f5ff681ae8f108cc5c532e419664355603 automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > /home/jenkins/git2/bin/git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git -c core.askpass=true fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1142/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a --contains 711b45729873bba2a021a1feaf618a316d783327 # timeout=10 > /home/jenkins/git2/bin/git rev-parse remotes/origin/pr/1142/merge^{commit} # timeout=10Checking out Revision 711b45729873bba2a021a1feaf618a316d783327 (origin/pr/1142/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f 711b45729873bba2a021a1feaf618a316d783327First time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1518/<h2>Build result: FAILURE</span></h2>GitHub pull request #1142 of commit 073cf9fb264116966547eb46bc7cf92674544d9f automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > /home/jenkins/git2/bin/git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > /home/jenkins/git2/bin/git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > /home/jenkins/git2/bin/git --version # timeout=10 > /home/jenkins/git2/bin/git -c core.askpass=true fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=15 > /home/jenkins/git2/bin/git rev-parse origin/pr/1142/merge^{commit} # timeout=10 > /home/jenkins/git2/bin/git branch -a --contains 5b75185a8a9b69552fced6383e13828bafe7bc07 # timeout=10 > /home/jenkins/git2/bin/git rev-parse remotes/origin/pr/1142/merge^{commit} # timeout=10Checking out Revision 5b75185a8a9b69552fced6383e13828bafe7bc07 (origin/pr/1142/merge) > /home/jenkins/git2/bin/git config core.sparsecheckout # timeout=10 > /home/jenkins/git2/bin/git checkout -f 5b75185a8a9b69552fced6383e13828bafe7bc07First time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

ryan-williams commented:
I think the missing sbt-interface.jar has to do with `zinc` not being installed or configured properly on the Jenkins build machine. Not sure what to do about it, maybe I'll try to just turn off incremental-compilation, overall or just for Jenkins.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1519/<h2>Build result: FAILURE</span></h2>[...truncated 32 lines...]Triggering ADAM-prb ? 2.6.0,2.11,1.3.1,centosTriggering ADAM-prb ? 2.3.0,2.11,1.4.1,centosTriggering ADAM-prb ? 2.6.0,2.11,2.0.0,centosTriggering ADAM-prb ? 2.3.0,2.10,1.4.1,centosTriggering ADAM-prb ? 2.6.0,2.11,1.6.1,centosADAM-prb ? 2.3.0,2.11,1.5.2,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.10,1.5.2,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.10,2.0.0,centos completed with result FAILUREADAM-prb ? 2.6.0,2.10,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.10,1.6.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.10,2.0.0,centos completed with result FAILUREADAM-prb ? 2.6.0,2.11,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.10,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.10,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.6.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,2.0.0,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.6.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,2.0.0,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,1.6.1,centos completed with result SUCCESSNotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

ryan-williams commented:
All the Spark 2.0.0 builds failed with:

```
mvn clean -Dhadoop.version=2.3.0 -Dspark.version=2.0.0
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=1g; support was removed in 8.0
[INFO] Scanning for projects...
[ERROR] The build could not read 4 projects -> [Help 1]
[ERROR]   
[ERROR]   The project org.bdgenomics.adam:adam-core-spark2_2.10:0.19.1-SNAPSHOT (/home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.3.0/SCALAVER/2.10/SPARK_VERSION/2.0.0/label/centos/adam-core/pom.xml) has 5 errors
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-misc-spark2_2.10:test-jar is missing. @ line 93, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-metrics-spark2_2.10:jar is missing. @ line 99, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-io-spark2_2.10:jar is missing. @ line 103, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-cli-spark2_2.10:jar is missing. @ line 107, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-intervalrdd-spark2_2.10:jar is missing. @ line 111, column 17
[ERROR]   
[ERROR]   The project org.bdgenomics.adam:adam-apis-spark2_2.10:0.19.1-SNAPSHOT (/home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.3.0/SCALAVER/2.10/SPARK_VERSION/2.0.0/label/centos/adam-apis/pom.xml) has 3 errors
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-misc-spark2_2.10:test-jar is missing. @ line 105, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-core-spark2_2.10:jar is missing. @ line 115, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-core-spark2_2.10:test-jar is missing. @ line 119, column 17
[ERROR]   
[ERROR]   The project org.bdgenomics.adam:adam-cli-spark2_2.10:0.19.1-SNAPSHOT (/home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.3.0/SCALAVER/2.10/SPARK_VERSION/2.0.0/label/centos/adam-cli/pom.xml) has 9 errors
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-misc-spark2_2.10:test-jar is missing. @ line 116, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-misc-spark2_2.10:jar is missing. @ line 122, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-io-spark2_2.10:jar is missing. @ line 126, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-cli-spark2_2.10:jar is missing. @ line 130, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.utils:utils-metrics-spark2_2.10:jar is missing. @ line 134, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-core-spark2_2.10:jar is missing. @ line 142, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-core-spark2_2.10:test-jar is missing. @ line 146, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-apis-spark2_2.10:jar is missing. @ line 152, column 17
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-apis-spark2_2.10:test-jar is missing. @ line 156, column 17
[ERROR]   
[ERROR]   The project org.bdgenomics.adam:adam-assembly-spark2_2.10:0.19.1-SNAPSHOT (/home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.3.0/SCALAVER/2.10/SPARK_VERSION/2.0.0/label/centos/adam-assembly/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.bdgenomics.adam:adam-cli-spark2_2.10:jar is missing. @ line 115, column 17
```

Any ideas?

heuermh commented:
The version for those dependencies should be 0.2.8, which is on Maven Central.  Are those variable substitutions coming from a profile?  We had trouble with that before.

heuermh commented on pom.xml:
```diff
@@ -282,9 +282,9 @@
           <javacArgs>
             <javacArg>-source</javacArg>
-            <javacArg>${java.version}</javacArg>
+            <javacArg>1.7</javacArg>
```
we require jdk8 because of the transitive dependency on htsjdk through Hadoop-BAM

fnothaft replied:
+1, needs to be changed. Although, is there a reason to move away from the `${java.version}` property here, if we're generally moving back towards using the parameters throughout the POM? I would prefer going back to `${java.version}` on this line.

ryan-williams replied:
Sorry, I got cornered into doing this for a reason that I didn't have paged in and meant to go back and try to reconstruct what that was.

I'm pretty sure it was related to my making the build use `zinc`; for some reason `1.8` was not acceptable in that setting, and I had to hard-code `1.7` here. I'll try to dig back and get the full state on it for posterity, but in the meantime I was going to give up on `zinc` in this PR anyway, since I didn't know how to get around the "`sbt-interface.jar` not found" problem on Jenkins discussed elsewhere on this PR (I believe due to `zinc` not being installed on Jenkins in some way), so I'll revert this bit momentarily. Thanks!

heuermh commented on pom.xml:
```diff
@@ -19,8 +19,8 @@
-    <!-- informative only, used in about template substitution -->
-    <scala.version>2.10</scala.version>
+    <scala.version>2.10.6</scala.version>
+    <scala.version.prefix>2.10</scala.version.prefix>
```
changing this also requires a change to https://github.com/bigdatagenomics/adam/blob/master/adam-cli/src/main/java-templates/org/bdgenomics/adam/cli/About.java#L28

ryan-williams replied:
Interesting; what is the contract with that file? Every Maven property must be declared there? Why?

ryan-williams replied:
OK, I looked around this a bit but don't see what change I need to make; help?

The templating-maven-plugin docs say it's useful "if you need for example to have things in that code replaced with some properties values."

`${scala.version}` is already exposed there (though not used anywhere in the project afaict) and I have no use for exposing `${scala.version.prefix}` there, if that's what you mean?

fnothaft replied:
I think it is OK (preferable, really) to have `About` print the full Scala version, as opposed to the prefix.

ryan-williams replied:
Ah that makes sense @heuermh, thanks for the explanation. Sounds like in the end we can just leave as is though, lmk if otherwise.

heuermh commented on pom.xml:
```diff
@@ -334,40 +334,40 @@
       <dependency>
         <groupId>org.bdgenomics.adam</groupId>
-        <artifactId>adam-core_2.10</artifactId>
+        <artifactId>adam-core_${scala.version.prefix}</artifactId>
```
I'm fairly certain there's a reason why we can't do this, I know because I tried to do it once.  Of course, I don't remember why it didn't work . . .

heuermh replied:
https://github.com/bigdatagenomics/utils/pull/69

fnothaft replied:
It's OK to use the properties in the `artifactId` for _dependencies_, it doesn't work to use properties in the `artifactId` for the POM itself. I know we've gone back and forth on this a lot, but I think it's actually preferable to do what @ryan-williams is doing here. I'll make a comment at the end with my reasoning.

ryan-williams commented:
> The version for those dependencies should be 0.2.8, which is on Maven Central. Are those variable substitutions coming from a profile? We had trouble with that before.

Those errors are modules' `<dependency>`s not declaring `<version>`s (because they should pick them up from the parent POM), but something about that resolution breaking down and Maven thinking they are just missing a `<version>` specification.

fnothaft commented on scripts/jenkins-test:
```diff
@@ -7,13 +7,15 @@ ADAM_MVN_TMP_DIR=$(mktemp -d -t adamTestMvnXXXXXXX)
 
 # variable declarations
-export JAVA_HOME=/usr/java/jdk1.8.0_60
+if [ -z "$JAVA_HOME" ]; then
```
This is a bit ugly, but we always need to overwrite `$JAVA_HOME` here due to Jenkins doing "shenanigans" with the Java version.  The short story is that Jenkins will always set `$JAVA_HOME` to a bad value, and we are powerless to stop it. There's a much longer and uninteresting backstory. Can you revert this back, or add some other logic to disable overwriting `$JAVA_HOME`?

ryan-williams replied:
Cool! I'll remove it here then altogether, 1m.

ryan-williams commented:
OK, I reverted the `java.version` bit and removed the `jenkins-test` `JAVA_HOME`-handling; I don't know why the build would avoid [the failure above](https://github.com/bigdatagenomics/adam/pull/1142#issuecomment-251500908) this time around, so am sort of expecting that to happen again, and suspect that it is related to the `scala.version.prefix` usage throughout the POMs, so let's see what happens there.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1521/<h2>Build result: FAILURE</span></h2>[...truncated 32 lines...]Triggering ADAM-prb ? 2.6.0,2.11,1.3.1,centosTriggering ADAM-prb ? 2.3.0,2.11,1.4.1,centosTriggering ADAM-prb ? 2.6.0,2.11,2.0.0,centosTriggering ADAM-prb ? 2.3.0,2.10,1.4.1,centosTriggering ADAM-prb ? 2.6.0,2.11,1.6.1,centosADAM-prb ? 2.3.0,2.11,1.5.2,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.10,1.5.2,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.10,2.0.0,centos completed with result FAILUREADAM-prb ? 2.6.0,2.10,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.10,1.6.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.10,2.0.0,centos completed with result FAILUREADAM-prb ? 2.6.0,2.11,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.10,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.10,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.6.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,2.0.0,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.6.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,1.3.1,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,2.0.0,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,1.6.1,centos completed with result SUCCESSNotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1522/
Test PASSed.

ryan-williams commented:
Cool, so it turns out I just hadn't updated the `sed` regexs in `move_to_spark2` to properly handle lines that now have `_${scala.version.prefix}` instead of `_2.10`.

I am not sure what to do about this "**fnothaft** requested changes" error I'm still seeing.

fnothaft commented:
LGTM! Unless anyone has objections, I will squash this down and merge in the AM.

fnothaft commented:
Thanks @ryan-williams! I've merged this manually as ad50f7d95e4f88fc2b242fc7c0c37beb8d4729fe, e8b0fcdd3feba937bb66cd72bbbdcad86039eeb1, cd488dfd1ace53bc3aec0e694def59b8ab2c1344, d242a265b60e1ab39d20b71b37eb14c5c704d054, a0730f36655d8fbd5359d84b5564c83c90ff7cf5, and 89942fee598426c12d5fd6531c2fd25d1f40b4ea. As always, thank you for taking the time to make a nice cleanup pass! It is greatly appreciated.
